blog/electron-25-0.md

```diff
@@ -0,0 +1,155 @@
+})
+```
+
+
+#### Deprecated: `BrowserWindow.setTrafficLightPosition(position)`
+
+`BrowserWindow.setTrafficLightPosition(position)` has been deprecated, the
+`BrowserWindow.setWindowButtonPosition(position)` API should be used instead
+which accepts `null` instead of `{ x: 0, y: 0 }` to reset the position to
+system default.
+
+```js
+// Deprecated in Electron 25
+win.setTrafficLightPosition({ x: 10, y: 10 })
+win.setTrafficLightPosition({ x: 0, y: 0 })
+
+// Replace with
+win.setWindowButtonPosition({ x: 10, y: 10 })
+win.setWindowButtonPosition(null)
+```
+
+#### Deprecated: `BrowserWindow.getTrafficLightPosition()`
+
+`BrowserWindow.getTrafficLightPosition()` has been deprecated, the
+`BrowserWindow.getWindowButtonPosition()` API should be used instead
+which returns `null` instead of `{ x: 0, y: 0 }` when there is no custom
+position.
+
+```js
+// Deprecated in Electron 25
+const pos = win.getTrafficLightPosition()
+if (pos.x === 0 && pos.y === 0) {
+  // No custom position.
+}
+
+// Replace with
+const ret = win.getWindowButtonPosition()
+if (ret === null) {
+  // No custom position.
+}
+```
+
+### New Features
+
+* Added BrowserWindow.set/getWindowButtonPosition APIs. [#37094](https://github.com/electron/electron/pull/37094)
+* Added `protocol.handle`, replacing and deprecating `protocol.{register,intercept}{String,Buffer,Stream,Http,File}Protocol`. [#36674](https://github.com/electron/electron/pull/36674) 
+* Added a `will-frame-navigate` event to `webContents` and the `<webview>` tag, which fires whenever any frame within the frame hierarchy attempts to navigate. [#34418](https://github.com/electron/electron/pull/34418)
+* Added initiator information to navigator events. This information allows distinguishing `window.open` from a parent frame causing a navigation, as opposed to a child-initiated navigation. [#37085](https://github.com/electron/electron/pull/37085)
+* Added net.resolveHost that resolves hosts using defaultSession object. [#38152](https://github.com/electron/electron/pull/38152)
+* Added new 'did-resign-active' event to `app`. [#38018](https://github.com/electron/electron/pull/38018)
+* Added several standard page size options to `webContents.print()`. [#37159](https://github.com/electron/electron/pull/37159)
+* Added the `enableLocalEcho` flag to the session handler `ses.setDisplayMediaRequestHandler()` callback for allowing remote audio input to be echoed in the local output stream when `audio` is a `WebFrameMain`. [#37315](https://github.com/electron/electron/pull/37315)
+* Added thermal management information to `powerMonitor`. [#38028](https://github.com/electron/electron/pull/38028)
+* Allows an absolute path to be passed to the session.fromPath() API. [#37604](https://github.com/electron/electron/pull/37604) 
+* Changed `net.fetch` to support requests to `file:` URLs and custom protocols registered with `protocol.register*Protocol`. [#36606](https://github.com/electron/electron/pull/36606) 
```

Comment:
Isn't `net.fetch` a completely new API for Electron 25? https://github.com/electron/electron/commit/872d1fe05af9bc14767a4c92f57dd6c8da80a400

We should document it much better and put it in front of the protocol deprecation section to introduce it in the first place.

Reply:
@georgexu99 and I added a blurb about fetch to a new section called "Highlights", also called out the protocol changes and Electron 22 lifetime extension